libs/payments/cart/src/lib/cart.utils.ts

```diff
@@ -41,6 +42,11 @@ export const cartEligibilityDetailsMap: Record<
     state: CartState.FAIL,
     errorReasonId: CartErrorReasonId.IAP_UPGRADE_CONTACT_SUPPORT,
   },
+  [EligibilityStatus.SAME]: {
+    eligibilityStatus: CartEligibilityStatus.INVALID,
```

Comment:
Note that this maps to `CartEligiblityStatus.INVALID`. IMO this is fine, since `SAME` is an invalid state, or more specifically can be considered a sub category of `INVALID`.

Reply:
Non-blocking, but I think keeping eligibilityStatus as a unique, narrowable field here will make it easier to narrow this type later.

Reply:
Would you mind elaborating a bit more? I'm not sure I'm following 100%